### PR TITLE
samples: net: Add overlay for nRF7002

### PR DESCRIPTION
**Pull request by rlubos**
Note - I've also considered having `boards/nrf7002dk_nrf5340_cpuapp.conf` in the samples, which works nicely with DK's, but this approach don't play well with EK configurations, i. e. an overlay would be needed anyway, so IMO it's better just to have a single copy of configs and use `overlay-nrf7002.conf` for DK and EK variants.

### CODE REVIEW

**Review by krish2718 — COMMENTED** [review]
is `lwm2m_client` missing?

**Comment by rlubos** [comment]
> is `lwm2m_client` missing?

This sample was not on the request list.

**Review by glarsennordic — APPROVED** [review]

File: include/zephyr/net/net_if.h [hunk 1]
```diff
@@ -1208,6 +1208,29 @@ bool net_if_ipv6_addr_rm(struct net_if *iface, const struct in6_addr *addr);
...
+ * @brief Go through all IPv6 addresses on a network interface and call callback
+ * for each used address.
+ *
+ * @param iface Pointer to the network interface
+ * @param cb User-supplied callback function to call
+ * @param user_data User specified data
+ */
+void net_if_ipv6_addr_foreach(struct net_if *iface, net_if_ip_addr_cb_t cb,
+			      void *user_data);
+
```
[the remark this answers is not in the document]
Comment (rlubos) [reply include/zephyr/net/net_if.h:1233]: Yes, it is: https://github.com/zephyrproject-rtos/zephyr/pull/60352

File: samples/net/dns_resolve/src/main.c [hunk 2]
```diff
@@ -152,57 +152,91 @@ static void do_ipv4_lookup(struct k_work *work)
...
 static void setup_dhcpv4(struct net_if *iface)
 {
+	bool found;
+
+	net_if_ipv4_addr_foreach(iface, check_dhcpv4_addr, &found);
+
+	if (found) {
+		/* Already have DHCP assigned address, schedule queries. */
+		schedule_ipv4_queries();
+		return;
```
[the remark this answers is not in the document]
Comment (rlubos) [reply samples/net/dns_resolve/src/main.c:237]: Yes, the sample is really simple - it just sends DNS queries once and that's it, I wouldn't complicate it.

File: samples/net/dns_resolve/src/main.c [hunk 3]
```diff
@@ -152,57 +152,91 @@ static void do_ipv4_lookup(struct k_work *work)
...
+	}
+
+	*found = true;
 }
 
 static void setup_dhcpv4(struct net_if *iface)
 {
+	bool found;
+
+	net_if_ipv4_addr_foreach(iface, check_dhcpv4_addr, &found);
```
[the remark this answers is not in the document]
Comment (rlubos) [reply samples/net/dns_resolve/src/main.c:232]: Thanks, I'll apply your suggestions in upstream PR

[the remark this answers is not in the document]
Comment (glarsennordic) [reply samples/net/dns_resolve/src/main.c:232]: Looks great, thanks!

**Pull request merged**